lib/model/query/user-preferences.js

```diff
@@ -38,24 +38,36 @@ const writeSiteProperty = (userId, propertyName, propertyValue) => ({ run }) =>
       UPDATE SET "propertyValue"=EXCLUDED."propertyValue"
 `);
 
+writeSiteProperty.audit = (userId, propertyName) => (log) =>
+  log('user.preference.update', null, { site: propertyName });
+
 const writeProjectProperty = (userId, projectId, propertyName, propertyValue) => ({ run }) => run(sql`
   INSERT INTO user_project_preferences ("userId", "projectId", "propertyName", "propertyValue")
     VALUES(${userId}, ${projectId}, ${propertyName}, ${JSON.stringify(propertyValue)})
     ON CONFLICT("userId", "projectId", "propertyName") DO
       UPDATE SET "propertyValue"=EXCLUDED."propertyValue"
 `);
 
+writeProjectProperty.audit = (userId, projectId, propertyName) => (log) =>
+  log('user.preference.update', null, { project: propertyName });
```

Comment:
Rather than differentiating site vs. project via the name of the property in `details`, how about using a fixed name for the property in `details`, then either logging `projectId` or not? Since we have the `projectId`, I think we might as well log it, since it's how we group preferences together.

```suggestion
  log('user.preference.update', null, { name: propertyName, projectId, value: propertyValue });
```